main.cpp

```diff
@@ -137,6 +173,8 @@ FloatType& FloatType::operator/=(float f)
     return *this;
 }
 
+using DoubleFunctionPtr = void(*)(double&);
+
```

Comment:
This should be scoped to inside the DoubleType class.  it should not be a global alias.
- [ ] move alias.